app/precompiled.py

```diff
@@ -234,9 +225,55 @@ def overlay_template_areas(src_pdf, page_number=None, overlay=True):
     return png_from_pdf(pdf, page_number)
 
 
-def get_invalid_pages(src_pdf):
-    pdf_to_validate = _overlay_printable_areas(src_pdf)
-    return list(_get_out_of_bounds_pages(PdfFileReader(pdf_to_validate)))
+def get_invalid_pages_with_message(src_pdf):
+    message = ""
+    invalid_pages = []
+    are_pages_A4_portrait_oriented, invalid_pages = _are_pages_A4_portrait_oriented(src_pdf)
+    if not are_pages_A4_portrait_oriented:
+        message = 'The page orientation is landscape instead of portrait on '
+    else:
+        pdf_to_validate = _overlay_printable_areas(src_pdf)
+        invalid_pages = list(_get_out_of_bounds_pages(PdfFileReader(pdf_to_validate)))
+        if len(invalid_pages) > 0:
+            message = 'Content in this PDF is outside the printable area on '
+    if len(invalid_pages) > 0:
+        message += unescaped_formatted_list(
+            invalid_pages,
+            before_each='',
+            after_each='',
+            prefix='page',
+            prefix_plural='pages'
+        )
+    return (invalid_pages, message)
+
+
+def _are_pages_A4_portrait_oriented(src_pdf):
+    pdf = PdfFileReader(src_pdf)
+    invalid_pages = []
+    for page_num in range(0, pdf.numPages):
+        page = pdf.getPage(page_num)
+
+        # page size is in points, there are 72 points in an inch and 25.4 milimeters in an inch,
+        # hence the transformations below to get size in milimeters
+        page_height = float(page.mediaBox.getHeight()) / 72 * 25.4
+        page_width = float(page.mediaBox.getWidth()) / 72 * 25.4
```

Comment:
i think it would be good to use the units included in reportlab - `mediabox.getHeight() / mm` should do it i think

Reply:
I don't think I understand... The current calculations give us size in milimeters, which is standard size of measuring A4, no?

Reply:
yeah, the calculation is correct (to go from points to mm), but rather than having magic numbers and doing the calculation ourselves, we've already got some shortcuts. elsewhere in precompiled.py we've got statements like `SOME_MEASUREMENT * mm` - `mm` is a hardcoded definition provided by `reportlab.units`, a module with a few hardcoded definitions for conversion

Reply:
ahhh, neat :3

Reply:
thanks, I've applied that now ☺️